File: Sources/cpp/src/linear_movement.cpp
#include "linear_movement.h"

#include <iostream>

#include "geometry.h"

void LinearMovement::update(long timeSinceLastUpdate, Entity * entity) {
    auto offset = entity->direction * entity->speed * timeSinceLastUpdate * 0.001;
    auto updatedFrame = entity->frame.offset(offset);
    entity->frame = updatedFrame;
}

File: Sources/windows/App/Sources/Capabilities/LinearMovement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace App
{
    public class LinearMovement : EntityCapability
    {
        public override void Update(TimeSpan timeSinceLastUpdate, Entity entity)
        {
            var frameDuration = timeSinceLastUpdate.TotalMilliseconds;
            var offset = entity.Direction * entity.Speed * frameDuration * 0.001;
            var updatedFrame = entity.Frame.Offset(offset);
            entity.Frame = updatedFrame;
        }
    }
}
